source/apiVolontaria/volunteer/models.py

```diff
@@ -227,6 +240,18 @@ class Meta:
     event = models.ForeignKey(Event, related_name='participation')
     user = models.ForeignKey(User, related_name='participation')
 
+    presence_duration = models.DecimalField(
+        decimal_places=2,
+        max_digits=10,
+        default=0
+    )
```

Comment:
It's not easy to know what i can put into this field ?
It's in Hours ? Days ? Minutes ?

Reply:
Why put `0` as default in place of `null` ?
We don't know so it's `null`, `0` can be `0` and can be misleading

Reply:
I suggest adding an help_text
In my opinion, it has to be in hour for simplicity for the average user.
Maybe renaming the field to presence_duration_hour ?

Reply:
Average user doesn't read the code, they only use the frontend and it's not our direct problem.
I think it's better to use a standard int type so it's easy to integrate for people who use the API.

I think we can use a `PositiveIntegerField` set as `null` by default that represent the number of minutes (the smallest unit of time we can use)
 -

Reply:
I agree from a developper point of view. But entering 5h and a quarter is a lot easier to enter 5,25 then 325.  We don't want the user to use a calculator each time they enter a value.  I'm not sure how I would make this work in the admin panel (convert 5,25 to 325 directly in the list view).

Reply:
If you want to enter 5H15 i'm ok. But after that when you want 5H10 it's not the same joke :p

`5 * 60 + 10` it's more easy than `5 + 10 / 60`. And it's more accurate since `5 + 10 / 60` will be `5.1667` and will be truncated or rounded most of the time.

The frontend need to transform that for each type of people, but DB need an accurante and easy to work format.

Have you already worked with [Django's DurationField](https://docs.djangoproject.com/en/1.11/ref/models/fields/#durationfield) ?

Reply:
The DurationField seem to have a problem with sqlite.  Here my output when trying to migrate : 
...
File "/Users/yanic/workspace/nousrire/API-env/lib/python3.6/site-packages/django/db/models/fields/__init__.py", line 1657, in get_db_prep_value
    return int(round(value.total_seconds() * 1000000))
AttributeError: 'int' object has no attribute 'total_seconds'

Here the source : https://docs.djangoproject.com/fr/1.11/_modules/django/db/models/fields/#DurationField

If it's no go for you in Decimal, I'll revert to PositiveIntegerField and try something for the admin...

Reply:
I checked a little on the web and some people seems to have problem with the `DurationField`, since it's not very well documented. If you're sure about your problem i guess `PositiveIntegerField` is the best option.

Decimal is definitively not the good way to do it :)